test/integration/smoke/test_primary_storage_nfsopts_kvm.py

```diff
@@ -0,0 +1,162 @@
+            state='Up',
+            id=None,
+        )
+        if response and len(response) > 0:
+            self.storage_pool = response[0]
+            return self.storage_pool
+        raise self.skipTest("Not enough KVM hosts found, skipping NFS options test")
+
+    def getNFSOptionsFromVirsh(self, poolId):
+        virsh_cmd = "virsh pool-dumpxml %s" % poolId
+        xml_res = self.sshClient.execute(virsh_cmd)
+        xml_as_str = ''.join(xml_res)
+        self.debug(xml_as_str)
+        parser = etree.XMLParser(remove_blank_text=True)
+        root = ET.fromstring(xml_as_str, parser=parser)
+        mount_opts = root.findall("{http://libvirt.org/schemas/storagepool/fs/1.0}mount_opts")[0]
+
+        options_map = {}
+        for child in mount_opts:
+            option = child.get('name').split("=")
+            options_map[option[0]] = option[1]
+        return options_map
+
+    def getUnusedNFSVersions(self, filter):
+        nfsstat_cmd = "nfsstat -m | sed -n '/%s/{ n; p }'" % filter
+        nfsstats = self.sshClient.execute(nfsstat_cmd)
+        versions = {"4.1": 0, "4.2": 0, "3": 0}
+        for stat in nfsstats:
+            vers = stat[stat.find("vers"):].split("=")[1].split(",")[0]
+            versions[vers] += 1
+
+        for key in versions:
+            if versions[key] == 0:
+                return key
+        return None
+
+    def getNFSOptionForPool(self, option, poolId):
+        nfsstat_cmd = "nfsstat -m | sed -n '/%s/{ n; p }'" % poolId
+        nfsstat = self.sshClient.execute(nfsstat_cmd)
+        if (nfsstat == None):
+            return None
+        stat = nfsstat[0]
+        vers = stat[stat.find(option):].split("=")[1].split(",")[0]
+        return vers
+
+    @attr(tags=["devcloud", "advanced", "advancedns", "smoke", "basic", "sg"], required_hardware="true")
+    def test_primary_storage_nfs_options_kvm(self):
+        """
+            Tests that NFS mount options configured on the primary storage are set correctly on the KVM hypervisor host
+        """
+
+        vers = self.getNFSOptionForPool("vers", self.storage_pool.id)
+        if (vers == None):
+            raise self.skipTest("Storage pool not associated with the host")
+
+        version = self.getUnusedNFSVersions(self.storage_pool.ipaddress)
+        nconnect = None
+        if version == None:
+            self.debug("skipping nconnect mount option as there are multiple mounts already present from the nfs server for all versions")
+            version = self.getUnusedNFSVersions(self.storage_pool.id)
+            nfsopts = "vers=" + version
+        else:
+            nconnect='4'
+            nfsopts = "vers=" + version + ",nconnect=" + nconnect
+
+        details = [{'nfsopts': nfsopts}]
+
+        maint_cmd = enableStorageMaintenance.enableStorageMaintenanceCmd()
+        maint_cmd.id = self.storage_pool.id
+        resp = self.apiclient.enableStorageMaintenance(maint_cmd)
+
+        storage = StoragePool.update(self.apiclient,
```

Comment:
as this test actually changes the environment, I think it disqualifies as a smoke test and should be a component test, actually under the maint sub package.
also it does not restore the original (lack of) options

Reply:
@DaanHoogland I can update the test so that it reverts the environment back to the original state.
What in general defines a smoke test vs a component test?